zebra-consensus/src/checkpoint.rs

```diff
@@ -56,10 +56,23 @@ struct QueuedBlock {
 
 /// A list of unverified blocks at a particular height.
 ///
-/// Typically contains zero or one blocks, but might contain more if a peer
+/// Typically contains a single block, but might contain more if a peer
 /// has an old chain fork. (Or sends us a bad block.)
+///
+/// The CheckpointVerifier avoids creating zero-block lists.
 type QueuedBlockList = Vec<QueuedBlock>;
 
+/// The maximum number of queued blocks at any one height.
+///
+/// This value is a tradeoff between:
+/// - rejecting bad blocks: higher is more efficient, and
+/// - avoiding a memory DoS: lower means less memory usage.
+///
+/// If we use a checkpoint interval of 500 blocks, our approximate queued block
+/// memory usage is:
+/// `500 blocks * 2 MB max per block * 2 blocks per height = 2 GB`
```

Comment:
Although this is an upper bound on our memory use, I don't think that this is a reasonable approximation for queued block memory usage, because it assumes that we only communicate with malicious peers who feed us giant blocks.

Reply:
I deleted this comment in 65ee727c8c3f87dc2ef539279684a0abb2d56dc8.

When I create the checkpoint list, I'll update that comment for block-size-based checkpoint intervals.